pkg/oauth/apiserver/apiserver.go

```diff
@@ -95,49 +144,103 @@ func (c *completedConfig) newV1RESTStorage() (map[string]rest.Storage, error) {
 		saAccountGrantMethod = oauthapiv1.GrantHandlerType(c.ExtraConfig.ServiceAccountMethod)
 	}
 
-	oauthClient, err := oauthclient.NewForConfig(c.GenericConfig.LoopbackClientConfig)
-	if err != nil {
-		return nil, err
-	}
 	routeClient, err := routeclient.NewForConfig(c.kubeAPIServerClientConfig)
 	if err != nil {
-		return nil, err
-	}
-	coreV1Client, err := corev1.NewForConfig(c.kubeAPIServerClientConfig)
-	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	combinedOAuthClientGetter := oauthserviceaccountclient.NewServiceAccountOAuthClientGetter(
-		coreV1Client,
-		coreV1Client,
-		coreV1Client.Events(""),
+		corev1Client,
+		corev1Client,
+		corev1Client.Events(""),
 		routeClient,
-		oauthClient.OAuthClients(),
+		oauthClient.OauthV1().OAuthClients(),
 		saAccountGrantMethod,
 	)
 	authorizeTokenStorage, err := authorizetokenetcd.NewREST(c.GenericConfig.RESTOptionsGetter, combinedOAuthClientGetter)
 	if err != nil {
-		return nil, fmt.Errorf("error building REST storage: %v", err)
+		return nil, nil, fmt.Errorf("error building REST storage: %v", err)
 	}
 	accessTokenStorage, err := accesstokenetcd.NewREST(c.GenericConfig.RESTOptionsGetter, combinedOAuthClientGetter)
 	if err != nil {
-		return nil, fmt.Errorf("error building REST storage: %v", err)
+		return nil, nil, fmt.Errorf("error building REST storage: %v", err)
 	}
 	clientAuthorizationStorage, err := clientauthetcd.NewREST(c.GenericConfig.RESTOptionsGetter, combinedOAuthClientGetter)
 	if err != nil {
-		return nil, fmt.Errorf("error building REST storage: %v", err)
+		return nil, nil, fmt.Errorf("error building REST storage: %v", err)
 	}
 	userOAuthAccessTokensDelegate, err := useroauthaccesstokensdelegate.NewREST(accessTokenStorage)
 	if err != nil {
-		return nil, fmt.Errorf("error building REST storage: %v", err)
+		return nil, nil, fmt.Errorf("error building REST storage: %v", err)
 	}
+	tokenReviewStorage, tokenReviewPostStartHooks, err := c.tokenReviewStorage(corev1Client, oauthClient, userClient)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error building REST storage: %v", err)
+	}
+
+	v1Storage := map[string]rest.Storage{
+		"oAuthAuthorizeTokens":      authorizeTokenStorage,
+		"oAuthAccessTokens":         accessTokenStorage,
+		"oAuthClients":              clientStorage,
+		"oAuthClientAuthorizations": clientAuthorizationStorage,
+		"userOAuthAccessTokens":     userOAuthAccessTokensDelegate,
+		"tokenReviews":              tokenReviewStorage,
+	}
+	return v1Storage, tokenReviewPostStartHooks, nil
+}
+
+func (c *completedConfig) tokenReviewStorage(
```

Comment:
@sttts this is the meat.  It needs to be perfectly congruent to https://github.com/openshift/kubernetes/blob/master/pkg/kubeapiserver/authenticator/patch_authenticator.go#L29